File: services/flight-service/model/flight.go
package model

import "time"

type Flight struct {
	ID             int64     `json:"id"`
	FlightNumber   string    `json:"flightNumber"`
	Origin         string    `json:"origin"`
	Destination    string    `json:"destination"`
	DepartureTime  time.Time `json:"departureTime"`
	ArrivalTime    time.Time `json:"arrivalTime"`
	Status         string    `json:"status"`
	BasePrice      float64   `json:"basePrice"`
	Currency       string    `json:"currency"`
	AvailableSeats int       `json:"availableSeats"`
	DurationMinutes int      `json:"durationMinutes"`
}

type FlightSearchResponse struct {
	Flights []Flight `json:"flights"`
}

File: app/web/lib/types/flight.ts
// Statuses the qoomlee API can return; flight search only returns SCHEDULED
// (see API_SPECS.md and flight/search_repository.go).
export type FlightStatus = "SCHEDULED" | "DELAYED" | "CANCELLED";

export interface Flight {
  id: number;
  flightNumber: string;
  origin: string;
  destination: string;
  departureTime: string; // ISO 8601
  arrivalTime: string; // ISO 8601
  basePriceMinor: number;
  currency: string;
  availableSeats: number;
  status: FlightStatus;
  durationMinutes: number;
}
